### ISSUE TITLE
Null user identity when flushing sql stats cache

### ISSUE THREAD

**Issue by at055612** [gchq/stroom#1464]
```
ERROR 2020-04-21T10:26:18.630Z ForkJoinPool.commonPool-worker-3 s.l.impl.LifecycleService      Task has null user identity: Flush SQL Statistic Cache 
java.lang.NullPointerException: Task has null user identity: Flush SQL Statistic Cache
	at java.base/java.util.Objects.requireNonNull(Objects.java:246)
	at stroom.task.impl.TaskContextImpl.<init>(TaskContextImpl.java:43)
	at stroom.task.impl.TaskContextFactoryImpl.wrap(TaskContextFactoryImpl.java:84)
	at stroom.task.impl.TaskContextFactoryImpl.createFromFunction(TaskContextFactoryImpl.java:71)
	at stroom.task.impl.TaskContextFactoryImpl.createFromConsumer(TaskContextFactoryImpl.java:63)
	at stroom.task.impl.TaskContextFactoryImpl.context(TaskContextFactoryImpl.java:42)
	at stroom.statistics.impl.sql.SQLStatisticCacheImpl.doFlush(SQLStatisticCacheImpl.java:126)
	at stroom.statistics.impl.sql.SQLStatisticCacheImpl.flush(SQLStatisticCacheImpl.java:115)
	at stroom.statistics.impl.sql.SQLStatisticCacheImpl.shutdown(SQLStatisticCacheImpl.java:161)
	at stroom.lifecycle.api.RunnableWrapper.run(RunnableWrapper.java:12)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run$$$capture(CompletableFuture.java:1807)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1799)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**Comment by at055612** [comment]
This is still an issue in beta.28

**Comment by at055612** [comment]
Was fixed in 394be6a59a3ce48e225ff13c1888014e889913af

**Comment by at055612** [comment]
Available in beta.30
